src/server/lib/okta/api/sessions.ts

```diff
@@ -33,6 +33,29 @@ export const getSession = async (
   }).then(handleSessionResponse);
 };
 
+/**
+ * Close a User session by session Id
+ *
+ * https://developer.okta.com/docs/reference/api/sessions/#close-session
+ *
+ * The Okta API closes the users session on success or
+ * returns a 404 on invalid.
+ *
+ * @param sessionId Okta session ID
+ * @returns Promise<boolean>
+ */
+export const closeSession = async (sessionId: string): Promise<undefined> => {
+  const path = buildUrl('/api/v1/sessions/:sessionId', { sessionId });
+  const response = await fetch(joinUrl(okta.orgUrl, path), {
+    method: 'DELETE',
+    headers: { ...defaultHeaders, ...authorizationHeader() },
+  });
+
+  if (!(response.ok || response.status === 404)) {
+    return handleErrorResponse(response);
```

Comment:
the first part of this logic makes sense to me, as in "if response.ok is falsy, handle error response".

But the second part I think is saying "if response.status is not 404", which is confusing me.. what if the status is 200, why would that warrant handling an error response?

Reply:
The endpoint will close the session ID passed to it, *and* if the session we asked it to close doesn't exist/is invalid, it will return 404. Otherwise it'll pass `204 No Content`. So 404 isn't an error code we need to know about - if the session doesn't exist, it doesn't need to be closed.

`response.ok` is returned `true` from `fetch()` on [a status in the range 200-299](https://developer.mozilla.org/en-US/docs/Web/API/Response/ok). So here we're saying "if the response is any code 0-199 and 300-infinity but not 404, return an error".

I can add some of this in a comment above this line!

Reply:
Could be clearer to write `if (response.ok === false && response.status != 404)`.

Reply:
it does look clearer, having the external logical NOT made it harder to read
yep— that comment idea would be great just because it wasn't immediately obvious